mlrun/package/context_handler.py

```diff
@@ -323,3 +305,52 @@ def _collect_mlrun_packagers(self):
                 is_mandatory=False,
                 is_custom_packagers=False,
             )
+
+    def _is_logging_worker(self):
+        """
+        Check if the current worker is the logging worker.
+
+        :return: True if the context belongs to the logging worker and False otherwise.
+        """
+        # If it's a OpenMPI job:
+        if self._context.labels.get("kind", "job") == "mpijob":
+            from mpi4py import MPI
+
+            from mlrun import mlconf
+
+            # Get the global and compare to the logging rank (worker) set in MLRun's configuration:
+            comm = MPI.COMM_WORLD
+            return comm.Get_rank() == mlconf.package.logging_worker
+
+        # Single worker is always the logging worker:
+        return True
+
+    def _validate_objects_to_log_hints_length(
+        self,
+        outputs: list,
+        log_hints: List[Union[Dict[str, str], str, None]],
+    ):
+        """
+        Validate the outputs and log hints are the same length. If they are not, warnings will be printed on what will
+        be ignored.
+
+        :param outputs:   List of outputs to log.
+        :param log_hints: List of log hints (logging configurations) to use.
+        """
+        if len(outputs) != len(log_hints):
+            self._context.logger.warn(
+                f"The amount of outputs objects returned from the function ({len(outputs)}) does not match the amount "
+                f"of provided log hints ({len(log_hints)})."
+            )
```

Comment:
@liranbg told me that already and when I did it, it failed horribly in the CI... I don't remember why though, maybe he does.

Reply:
not ci, but perhaps nuclio. ever since, iirc Ive fixed the logger to be able to log kwargs (it used a log that failed to use kwargs)
so alon comment is still relevant

Reply:
will do it for the entire `mlrun.package` module on another PR.